src/main/java/org/mockito/junit/VerificationCollectorImpl.java

```diff
@@ -20,26 +21,36 @@
     private StringBuilder builder;
     private int numberOfFailures;
 
-    public VerificationCollectorImpl() {
+    VerificationCollectorImpl() {
         this.resetBuilder();
+        MOCKITO_CORE.setVerificationState(new VerificationStrategy() {
+            public VerificationMode verifyLazily(VerificationMode mode) {
+                return new VerificationWrapper(mode);
+            }
+        });
     }
 
     public Statement apply(final Statement base, final Description description) {
         return new Statement() {
             @Override
             public void evaluate() throws Throwable {
-                base.evaluate();
-                VerificationCollectorImpl.this.collectAndReport();
+                try {
+                    base.evaluate();
+                    VerificationCollectorImpl.this.collectAndReport();
+                } finally {
+                    // Guard against an exception thrown inside the testcase which was not from the collector
+                    Mockito.setVerificationState(MOCKITO_CORE.getDefaultVerificationStrategy());
+                }
             }
         };
     }
 
     public <T> T verify(T mock) {
```

Comment:
The previous implementation still required you to call `collector.verify`. That is this method. Same as method below.